virttest/qemu_vm.py

```diff
@@ -1252,6 +1252,15 @@ def set_value(opt_string, key, fallback=None):
 
             return secret_cmdline + " -spice %s" % (",".join(spice_opts))
 
+        def add_thread_context(devices, params):
+            for thread_context in params.objects("vm_thread_contexts"):
+                thread_context_params = params.object_params(thread_context)
+                dev = devices.thread_context_define_by_params(thread_context_params, thread_context)
+                set_cmdline_format_by_cfg(dev, self._get_cmdline_format_cfg(),
+                                          "thread_context")
```

Comment:
```suggestion
                                          "thread_contexts")
```
I think it should be plural to correspond to the `vm_thread_contexts`
CC @nickzhq

Reply:
Updated now, thanks!

Reply:
Disagree, here the name is device type, a type should not be plural.

Reply:
> Disagree, here the name is device type, a type should not be plural.

I remember that it is not just related to a device type but relates to a vt object, like, `images`, `monitors`, `mem_devs`, and so on.
Here we set the cmd line format base on the object `vm_thread_contexts` which is defined by vt.

Hi @nickzhq 
Please correct me if I was wrong since you developed this part.
Thanks.

Reply:
> vm_thread_contexts

Hello, 
in the beginning of vt json format design, the vt object is preferred instead device type. Among `images`, `monitors`, `mem_devs` are the concepts of vt framework.
For my opinion, use the object which is defined by vt.

Thanks!
cc @PaulYuuu @mcasquer

Reply:
> > vm_thread_contexts
> 
> Hello @YongxueHong , in the beginning of vt json format design, the vt object is preferred instead device type. Among `images`, `monitors`, `mem_devs` are the concepts of vt framework. For my opinion, use the object which is defined by vt.
> 
> Thanks! cc @PaulYuuu @mcasquer

OK, if it is VT type, then this is good to me, but we still have a mix in the current VT code, like `tpm` vs `tpms`
https://github.com/avocado-framework/avocado-vt/blob/7c6bf78d347f7851368f7ae0c493e2e035e21dc4/virttest/qemu_vm.py#L2670-L2677